File: Baekjoon/문제풀이/에디터/Yuri.py
import re

def L(str, num):
    if str.find("△")==-1:
       cursor = num - 1
       str = str[0:cursor-1]+"△"+str[cursor:]

    elif str.find("△")==0:
       return
    
    else:
        now_cursor = str.find("△")
        cursor = now_cursor - 1
        re.sub("△", "", str)
        str = str[0:cursor-1]+"△"+str[cursor:]

    

def D(str, num):
    if str.find("△")==-1 or str.find("△")==0:
       cursor = num
    
    else:
        now_cursor = str.find("△")
        cursor = now_cursor + 1
        re.sub("△", "", str)
        str = str[0:cursor]+"△"+str[cursor+1:]


def B(str, num):
   if str.find("△")==-1:
       str = str[0:num-1]
       cursor = num-1

   elif str.find("△")==0:
       return
    
   else:
        now_cursor = str.find("△")
        cursor = now_cursor - 1
        str = str[0:now_cursor-2]+"△"+str[now_cursor:]

def P(str, num, text):
   if str.find("△")==-1:
       str = str + text + "△"
       cursor = num+1

   elif str.find("△")==0:
       str = text + str
       cursor = 1
    
   else:
        now_cursor = str.find("△")
        cursor = now_cursor + 1
        str = str[0:now_cursor-1]+text+str[now_cursor:]

      

   str = input()
   T = int(input())  # 테스트 케이스 개수

   for _ in range(T):
       command = input()

       if command == D:
           D(str,len(str)-1)
      
       elif command == L:
            L(str,len(str)-1)

       elif command == B:
            B(str,len(str)-1)

       else:
            if str.find("P")==-1:
               "해당 명령어는 존재하지 않음"
            else:
               text=str.split()[1]
               P(str, len(str)-1, text)

   print(str)     

File: Baekjoon/문제풀이/에디터/yuri.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // 입력 문자열과 커서(△) 추가
        String str = sc.nextLine() + "△";
        int T = Integer.parseInt(sc.nextLine()); // 테스트 케이스 개수

        for (int i = 0; i < T; i++) {
            String[] command = sc.nextLine().split(" ");
            if (command[0].equals("D")) {
                str = D(str);
            } else if (command[0].equals("L")) {
                str = L(str);
            } else if (command[0].equals("B")) {
                str = B(str);
            } else if (command[0].equals("P")) {
                String text = command[1];
                str = P(str, text);
            } else {
                System.out.println("해당 명령어는 존재하지 않음");
            }
        }

        // 최종 문자열에서 커서(△) 제거 후 출력
        System.out.println(str.replace("△", ""));
    }

    // 커서를 왼쪽으로 이동
    public static String L(String str) {
        int cursorIdx = str.indexOf("△");
        if (cursorIdx > 0) {
            str = str.substring(0, cursorIdx - 1) + "△" + str.substring(cursorIdx - 1, cursorIdx) + str.substring(cursorIdx + 1);
        }
        return str;
    }

    // 커서를 오른쪽으로 이동
    public static String D(String str) {
        int cursorIdx = str.indexOf("△");
        if (cursorIdx < str.length() - 1) {
            str = str.substring(0, cursorIdx) + str.substring(cursorIdx + 1, cursorIdx + 2) + "△" + str.substring(cursorIdx + 2);
        }
        return str;
    }

    // 커서 왼쪽 문자를 삭제
    public static String B(String str) {
        int cursorIdx = str.indexOf("△");
        if (cursorIdx > 0) {
            str = str.substring(0, cursorIdx - 1) + "△" + str.substring(cursorIdx + 1);
        }
        return str;
    }

    // 커서 위치에 문자 추가
    public static String P(String str, String text) {
        int cursorIdx = str.indexOf("△");
        str = str.substring(0, cursorIdx) + text + "△" + str.substring(cursorIdx + 1);
        return str;
    }
}
